src/lib/CMakeLists.txt

```diff
@@ -514,6 +514,8 @@ set(
     utils/template_type.hpp
     utils/timer.cpp
     utils/timer.hpp
+    utils/tracing/provider.h
```

Comment:
What is this file used for? On my machine, it is empty.

Reply:
When you want to trace with dtrace you need to create a header file, that defines the required tokens. This will be written into `provider.h`. Since this file is auto generated, it is not supposed to be commited. However, if the file is deleted, some includes will not work and raise errors during compilation.

Reply:
That's what I thought, but as mentioned, it is empty. What do I have to do to fill it?

Reply:
Any chance you mean `probe_definitions.hpp`?

Reply:
To fill the file, run `dtrace -h -s src/lib/utils/tracing/provider.d -o src/lib/utils/tracing/provider.h`
I don't mean `probe_definitions.hpp`

Reply:
Ah, right. I remember seeing that. After running that command, I have to recompile everything, correct? In that case, shouldn't that step be part of the cmake process, just like `probe_definitions.hpp`?